### ISSUE TITLE
Note on header-only library

### ISSUE THREAD

**Issue by boris-kolpackov** [lyrahgames/build2-tinyobjloader#1]
This is just FYI: it seems Debian builds this library source-full:

https://packages.debian.org/bookworm/libtinyobjloader2rc10

Not saying we must do the same since there could be complications on platforms that Debian doesn't care about, like Windows with its symbol exporting macros. So if upstream say they only test/guarantee to work for header-only version, then we should probably stick to that.

**Comment by lyrahgames** [comment]
Thank you for the information! ^^ You must have seen that I put a long comment about it in the `Issues and Notes` section of the `README.md`. If Debian offers it like that then it was probably best that I provided an explanation there. At first, I added the support for a source-full version but quickly decided to remove it again mainly due to the comment in the notes and the reasons you stated. Furthermore, after inspecting the CMake files of the upstream repository, I am quite sure these contain minor errors concerning the build of the shared library. The shared version is neither documented nor used/tested anywhere in the repository.

**Comment by lyrahgames** [comment]
Hi @boris-kolpackov,

I wanted to revisit testing the source-full build.
However, as was the case in [this issue in `libigl`](https://github.com/build2-packaging/libigl/issues/4), I’d still like to preserve the header-only mode as well.
My main question is about the typical or recommended approach here:

Should I

- provide two separate library targets — one for the source-full build and another for the header-only mode (similar to how unit testing frameworks often provide a variant with a main function),

or

- introduce a configuration variable, e.g. `config.libtinyobjloader.header_only`, that determines how the library is consumed?

Since this library doesn’t define a main function, the second approach seems more consistent with the idea of dependency negotiation for transitive dependencies during package consumption.

**Comment by boris-kolpackov** [comment]
Yes, I think the second approach is the way to go. In particular, with the first approach you could theoretically end up with two libraries that are linked to the same executable using different flavors of the same library, which is not going to end up well.
